packages/core/src/internal/proc.js

```diff
@@ -318,11 +297,11 @@ export default function proc(
           result.sagaStack = sagaStackToString(result.sagaStack)
         }
 
-        if (onError) {
-          onError(result)
+        if (env.onError) {
```

Comment:
we could be more explicit here and do `if (typeof env.onError === 'function')`

Reply:
I've add check for `onError` in runSaga, so it must be a function if pass `if (onError)` here.